angular/src/app/landing/downloadstatus/downloadstatus.component.ts

```diff
@@ -45,19 +44,18 @@ export class DownloadstatusComponent implements OnInit {
             if(ev.key == this.dataCartStatus.getName()){
                 this.dataCartStatus.restore();
             }
-
-            // Emit item IDs whose download status is 'completed' 
-            let keys = Object.keys(this.dataCartStatus.dataCartStatusItems).filter(key => this.dataCartStatus.dataCartStatusItems[key].downloadPercentage == 100);
-
-            this.downloadedKeys.emit(keys);
         }
     }
 
     /**
      * Get keys of dataCartStatus. The UI uses it to display download progress
      */
-    get getKeys(){
-        return Object.keys(this.dataCartStatus.dataCartStatusItems)
+    get getKeys() { 
+        if(this.inBrowser) {
+            return Object.keys(this.dataCartStatus.dataCartStatusItems);
+        }else{
+            return null;
```

Comment:
For some reason, my previous attempt at this comment did not show the line of code I was commenting on (`downloadstatus.component.ts`: 56).

I think it would be safer to return `[]` (an empty list) instead of `null`.